nipy/algorithms/clustering/gmm.py

```diff
@@ -217,8 +218,8 @@ def plot2D(x, my_gmm, z=None, with_dots=True, log_scale=False, mpaxes=None,
         if z is None:
             plt.plot(x[:, 0], x[:, 1], 'o')
         else:
-            hsv = plt.cm.hsv(range(256))
-            col = hsv[range(0, 256, 256 // int(z.max() + 1))]
+            hsv = plt.cm.hsv(list(range(256)))
+            col = hsv[list(range(0, 256, 256 // int(z.max() + 1)))]
```

Comment:
I agree, I would prefer to avoid indexing with `range`, because it's harder to see what numpy is going to do with a `range` in Python 3.  It looks like it does the right thing, but I don't know where that is guaranteed.